### PR TITLE
feat: Cooperations

### PR DESCRIPTION
**Pull request by Jonasdoubleyou**

### CODE REVIEW

**Comment by Jonasdoubleyou** [comment]
c.f. https://user-app-feat-cooperati-zrskmc.herokuapp.com/registration/helper?cooperation=self

File: src/pages/Registration.tsx [hunk 1]
```diff
@@ -99,51 +104,70 @@ const Registration: React.FC = () => {
...
+        )
+    );
+
+    const cooperationTag = new URL(window.location.toString()).searchParams.get('cooperation');
+
+    const cooperation = useMemo(() => {
+        const result = corpData?.cooperations?.find((it) => it.tag === cooperationTag);
+
+        if (cooperationTag && corpData && !result) {
+            throw new Error(`Invalid Cooperation Tag '${cooperationTag}'`);
```
Comment (dhenkel92) [inline src/pages/Registration.tsx:130]: /nit
Should be fine for our fast DHL change, but I think we generally should try to avoid crashing hard

**Review by dhenkel92 — APPROVED** [review]
Tested the change in the review app. LGTM 👍

File: src/pages/Registration.tsx (hunk 1 above)
Comment (Jonasdoubleyou) [reply src/pages/Registration.tsx:130]: Hmmm, I think a misconfiguration of this should rather be detected by us (this shows the issue reporter) than being ignored (and thus we might report wrong numbers)

Comment (dhenkel92) [reply src/pages/Registration.tsx:130]: That's true, but failing hard is different than explaining what happened. 
It's the first time the people are interacting with our page and they have no idea if it's down or if it's just a broken link.

Btw. should there be an option to disable the cooperation?

Comment (Jonasdoubleyou) [reply src/pages/Registration.tsx:130]: Fair enough, I'll build something for that

**Pull request merged**